pkg/render/logstorage.go

```diff
@@ -570,11 +632,38 @@ func (es elasticsearchComponent) podTemplate() corev1.PodTemplateSpec {
 	}
 
 	initContainers := []corev1.Container{initOSSettingsContainer}
-
 	annotations := es.cfg.TrustedBundle.HashAnnotations()
 	annotations[ElasticsearchTLSHashAnnotation] = rmeta.SecretsAnnotationHash(es.cfg.ElasticsearchUserSecret)
 	annotations[es.cfg.ElasticsearchKeyPair.HashAnnotationKey()] = es.cfg.ElasticsearchKeyPair.HashAnnotationValue()
 
+	if operatorv1.IsFIPSModeEnabled(es.cfg.Installation.FIPSMode) {
+		initKeystore := corev1.Container{
+			Name:  keystoreInitContainerName,
+			Image: es.esImage,
+			SecurityContext: &corev1.SecurityContext{
+				Privileged: ptr.BoolToPtr(false),
+			},
+			Env: []corev1.EnvVar{
+				{
+					Name: ElasticsearchKeystoreEnvName,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{Name: ElasticsearchKeystoreSecret},
+							Key:                  ElasticsearchKeystoreEnvName,
+						},
+					},
+				},
+				{
+					Name:  "ES_JAVA_OPTS",
+					Value: "--module-path /usr/share/bc-fips/",
+				},
+			},
+			Command: []string{"/usr/bin/env", "bash", "-c", fmt.Sprintf(KeystoreInitScript, ElasticsearchKeystoreEnvName, ElasticsearchKeystoreEnvName, ElasticsearchKeystoreEnvName)},
```

Comment:
I can't say I'm a fan of this, I don't think we should be generating these containers on the fly and I don't think the code should live in the operator (the operator is basically injecting code into the container).

If the contents of `KeystoreInitScript` are needed to run the container, then I think it should be a part of the container.